### ISSUE TITLE
[REVIEW]: PiSCAT: A Python Package for Interferometric Scattering Microscopy

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#4024]
**Submitting author:** @po60nani (<a href="http://orcid.org/0000-0002-0724-3158">Houman Mirzaalian Dastjerdi</a>)
**Repository:** <a href="https://github.com/SandoghdarLab/PiSCAT" target ="_blank">https://github.com/SandoghdarLab/PiSCAT</a>
**Version:** 0.1.20
**Editor:** @emdupre
**Reviewer:** @ziatdinovmax, @aquilesC
**Archive:** Pending

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/81fb6452cdd3d3f792d50a4e8763940c/status.svg)](https://joss.theoj.org/papers/81fb6452cdd3d3f792d50a4e8763940c)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/81fb6452cdd3d3f792d50a4e8763940c"><img src="https://joss.theoj.org/papers/81fb6452cdd3d3f792d50a4e8763940c/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/81fb6452cdd3d3f792d50a4e8763940c/status.svg)](https://joss.theoj.org/papers/81fb6452cdd3d3f792d50a4e8763940c)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/SandoghdarLab/PiSCAT" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@ziatdinovmax & @aquilesC, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @emdupre know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @ziatdinovmax
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/SandoghdarLab/PiSCAT">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/SandoghdarLab/PiSCAT">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@po60nani) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @aquilesC
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/SandoghdarLab/PiSCAT">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/SandoghdarLab/PiSCAT">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@po60nani) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @ziatdinovmax, @aquilesC it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #4024 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.29 s (418.5 files/s, 89978.3 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                         101           4152           2738          12292
Markdown                         7            330              0            964
Jupyter Notebook                 5              0           5035            432
reStructuredText                 6            159            130            123
DOS Batch                        1              9              1             27
CSS                              1              2              0             11
make                             1              4              7             10
YAML                             1              4              6              7
-------------------------------------------------------------------------------
SUM:                           123           4660           7917          13866
-------------------------------------------------------------------------------


Statistical information for the repository 'ebfb0353e28a7dbe1bf5f3a9' was
gathered on 2021/12/21.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Houman Mirzaalian Da            78         19770           1046           96.56
Matthias Baer                    1           215            137            1.63
Reza Gholami M                   2             5              5            0.05
rgholam                          1           380              0            1.76

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Houman Mirzaalian Da      18583           94.0          1.0                3.17
Matthias Baer               215          100.0          7.0                5.58
rgholam                     384          101.1          2.3                4.17
```

**Comment by emdupre** [comment]
@whedon generate pdf from branch JOSS-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04024/joss.04024/10.21105.joss.04024.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04024/joss.04024/10.21105.joss.04024.pdf) :page_facing_up: :point_left:

**Comment by emdupre** [comment]
@whedon check references from branch JOSS-paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch JOSS-paper
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1364/OE.401374 is OK

MISSING DOIs

- 10.1103/physrevlett.93.037401 may be a valid DOI for title: Detection and Spectroscopy of Gold Nanoparticles Using Supercontinuum White Light Confocal Microscopy
- 10.1021/acs.nanolett.9b01822 may be a valid DOI for title: Interferometric scattering microscopy: seeing single nanoparticles and molecules via Rayleigh scattering
- 10.1101/2021.08.16.456463 may be a valid DOI for title: Optimized analysis for sensitive detection and analysis of single proteins via interferometric scattering microscopy
- 10.1038/ncomms5495 may be a valid DOI for title: Direct optical sensing of single unlabelled proteins and super-resolution imaging of their binding sites
- 10.1021/acs.nanolett.8b02240.s001 may be a valid DOI for title: High-speed microscopy of diffusion in pore-spanning lipid membranes

INVALID DOIs

- None
```

**Comment by emdupre** [comment]
Hi everyone :wave: Thanks again @ziatdinovmax and @aquilesC for agreeing to review this submission ! The review will take place in this issue.

Whenever possible, please open relevant issues on the linked software repository (and cross-link them with this issue) rather than discussing them here. This helps to make sure that feedback is translated into actionable items to improve the software. If you aren't sure how to get started, please see the [Reviewing for JOSS guide](https://joss.readthedocs.io/en/latest/reviewer_guidelines.html) -- and, of course, feel free to reach out with any questions !

**Comment by emdupre** [comment]
@po60nani a few initial points on the JOSS paper itself, rather than the software :
- [ ] The [manuscript](https://github.com/openjournals/joss-papers/blob/joss.04024/joss.04024/10.21105.joss.04024.pdf) does not yet have a 'Statement of Need' section. You can find more information in the [submission guidelines](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain) on what the statement of need section should contain
- [ ] Several references cited in the paper are missing DOIs, with whedon suggesting possible additions in [this comment](https://github.com/openjournals/joss-reviews/issues/4024#issuecomment-999013573). Could you please confirm whether the automatically suggested DOIs are appropriate for your references, and either add or correct them ?

**Comment by po60nani** [comment]
@whedon generate pdf from branch JOSS-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04024/joss.04024/10.21105.joss.04024.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04024/joss.04024/10.21105.joss.04024.pdf) :page_facing_up: :point_left:

**Comment by po60nani** [comment]
@whedon check references from branch JOSS-paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch JOSS-paper
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1103/PhysRevLett.93.037401 is OK
- 10.1021/acs.nanolett.9b01822 is OK
- 10.1007/978-3-030-21722-8 is OK
- 10.1088/1361-6463/ac2f68 is OK
- 10.1038/ncomms5495 is OK
- 10.1021/acs.nanolett.8b02240 is OK
- 10.1364/OE.401374 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by po60nani** [comment]
@whedon generate pdf from branch JOSS-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04024/joss.04024/10.21105.joss.04024.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04024/joss.04024/10.21105.joss.04024.pdf) :page_facing_up: :point_left:

**Comment by po60nani** [comment]
@emdupre Thank you for taking the time to provide comments. We made changes to the manuscript in response to your comments.

- [ ] The manuscript does not yet have a 'Statement of Need' section. You can find more information in the submission guidelines on what the statement of need section should contain

- _The new section called 'Statement of Need' was added to the manuscript._

- [ ] Several references cited in the paper are missing DOIs, with whedon suggesting possible additions in this comment. Could you please confirm whether the automatically suggested DOIs are appropriate for your references, and either add or correct them ?

- _All DOIs were added to the bibliography.._

**Comment by whedon** [comment]
:wave: @aquilesC, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @ziatdinovmax, please update us on how your review is going (this is an automated reminder).

**Comment by aquilesC** [comment]
I am going through the program, and I'll be incrementally listing what I find (some are possible improvements, some are important things to address), I'll clean them up at the end of the process: 

It is GREAT that you provide the software and sample data to get started! Kudos! 

- [ ] Couldn't find a simple way of downloading sample data. The ``download_tutorial_data`` failed because of write access (was trying to write to ``/Tutorials`` instead of a relative path. When looked into the code, I found the link to the ownclodu zip file. This could be linked from the docs themselves. 
- [ ] A bit of documentation on the files I'll be downloading is also helpful. Right now I got two .raw files without contextual information, which means I can't use the GUI to open them (it asks for information that it does not discover on the file, and I have no quick way of telling what it actually is, like image width and heigh)

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
It looks like this paper includes an ORCID (0000-0003-0872-7098) that is probably not correct.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04024/joss.04024/10.21105.joss.04024.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04024/joss.04024/10.21105.joss.04024.pdf) :page_facing_up: :point_left:

**Comment by po60nani** [comment]
@whedon generate pdf from branch JOSS-paper

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.04024/joss.04024/10.21105.joss.04024.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.04024/joss.04024/10.21105.joss.04024.pdf) :page_facing_up: :point_left:

**Comment by po60nani** [comment]
@emdupre, Would you please let me know if you received my response to the comments?

**Comment by emdupre** [comment]
Hi @po60nani, yes, thank you for confirming on this ! 

I'm still waiting on review comments from @ziatdinovmax  to move forward. We're in touch, and I'm expecting his comments soon.

**Comment by ziatdinovmax** [comment]
My apologies for the delay in reviewing the package. I was able to install it successfully and ran examples following the instructions provided by the authors. Their manuscript is well-written: it explains the need for this package and clearly outlines its structure. At the end of the manuscript, the authors mention future plans such as GPU support and incorporation of DNN. I would expect there are people who may be willing to contribute, but the contributing guidelines seem to be absent. Can the authors add them to the README file? 

One other minor issue: I recommend specifying version requirements for the packages listed in setup.py (open-cv, networkx, etc) as some of the listed packages have undergone quite a few breaking changes between different releases.

**Comment by po60nani** [comment]
**I was able to install it successfully and ran examples following the instructions provided by the authors. Their manuscript is well-written: it explains the need for this package and clearly outlines its structure.**

- _We appreciate the reviewer's acknowledgment of our paper's and package's utility and effectiveness._

**At the end of the manuscript, the authors mention future plans such as GPU support and incorporation of DNN. I would expect there are people who may be willing to contribute, but the contributing guidelines seem to be absent. Can the authors add them to the README file?**

- _We've already added this section to the community guidelines, and we've also added a link to contributing in the [Readme](https://github.com/SandoghdarLab/PiSCAT/blob/main/README.md#contributing), based on the second reviewer's comments._

**One other minor issue: I recommend specifying version requirements for the packages listed in setup.py (open-cv, networkx, etc) as some of the listed packages have undergone quite a few breaking changes between different releases.**

- _In our repository, we added a file called requirements.txt that specifies version requirements. We also mentioned this in the contributing guidelines, along with the Python version we tested._

**Comment by emdupre** [comment]
Thank you, @po60nani, for your in-depth responses to the reviewers's feedback ! 

@ziatdinovmax, I noticed that you have already completed your reviewer checklist. Can you confirm that you are happy with the submission as-is ?

@aquilesC, there are still a few items that are not yet checked on your checklist, as these were raised as concerns in the initial round of review. Could you confirm if the authors have sufficiently addressed your concerns during revisions to check off these items and complete your reviewer checklist ?

**Comment by aquilesC** [comment]
I have finished reviewing the manuscript. The only concern I have is regarding the **references**, and I would ask @emdupre for guidance on the matter. 

The authors claim: 

> We would have to mention all dependencies and explain them in the text if we wanted to add a Trackpy citation, which would be problematic because we intended to keep the number of pages under four. 

At the same time, in the manuscript they say:

> PiSCAT provides several algorithms for localization and tracking of nanoparticles

They do implement several localisation algorithms, but only one for tracking and it is based on Trackpy. 

Generally speaking, the PiSCAT package has [23 dependencies](https://github.com/SandoghdarLab/PiSCAT/blob/main/requirements.txt), some with clear academic relevance (Astropy, Trackpy, PyQtGraph) and some larger ones with scientific relevance (numpy, SciPy, scikit, matplotlib). I wonder where the threshold to require a citation for software should be put. After all, there are 7 self-references, but none to software.

Once this is clarified *editorially* (also for my own reference for future reviews), I'll conclude the review.

**Comment by po60nani** [comment]
@aquilesC, Thanks for clarifying. We can add Trackpy references to the place where we mention tracking based on your explanation. You can find the updated version of the manuscript in the following.

**Comment by po60nani** [comment]
@editorialbot generate pdf from branch JOSS-paper

**Comment by po60nani** [comment]
@editorialbot set JOSS-paper as branch

**Comment by po60nani** [comment]
@editorialbot generate pdf

**Comment by emdupre** [comment]
Thank you for your feedback, @aquilesC, and your recent updates @po60nani ! I'm glad that compiling with the new `editorial-bot` system was smooth :slightly_smiling_face: The previous Whedon system has been deprecated, so note that all future compilation, etc, commands can be issued directly to editorial-bot for both this and any other JOSS submissions.

For clarity, I would just like to confirm that JOSS does not have a strict page limit. Although [we generally recommend](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain) that papers are at most approximately 1000 words, this is not a hard limit and can be adjusted to accommodate information that the authors or reviewers believe is necessary to understand the presented research software and its contribution to the field.

With that in mind, I agree with @aquilesC that citing the appropriate software is well-worth inclusion in the paper. I appreciate that the authors have added a reference to `TrackPy` in the "Concept and Structure of PiSCAT" section to point to supported localisation algorithms as a step in this direction. Considering citing other software dependencies:

> Generally speaking, the PiSCAT package has [23 dependencies](https://github.com/SandoghdarLab/PiSCAT/blob/main/requirements.txt), some with clear academic relevance (Astropy, Trackpy, PyQtGraph) and some larger ones with scientific relevance (numpy, SciPy, scikit, matplotlib). I wonder where the threshold to require a citation for software should be put. After all, there are 7 self-references, but none to software.

There is not a strict rule that all dependencies for a given package must be cited. Instead, this is generally an editorial judgment, with more significant dependencies for a project receiving acknowledgement in-text. For example, in the [example paper provided in the JOSS documentation](https://joss.readthedocs.io/en/latest/submitting.html#example-paper-and-bibliography), there is a reference to one other research software project, from which the example project significantly benefits:

> `Gala` also relies heavily on and interfaces well with the implementations of physical units and astronomical coordinate systems in the `Astropy` package [@\astropy] (`astropy.units` and `astropy.coordinates`).

This acknowledgement features in the Statement of Need section, which is generally used to overview the current landscape of the field and how the present package contributes to it. While the authors provide a thorough and effective scope of PiSCAT in the current paper's Statement of Need, it would likely help orient readers if they could directly note there any other research software whose functionality significantly supports or overlaps with that of PiSCAT. It is likely that TrackPy merits mention here, given the usage of its tracking algorithm. 

I hope this helps to clarify your questions around references, @aquilesC, and provides relevant context for making appropriate edits, @po60nani ! I have several other, minor edits on the paper and reference formatting which I will include in my next comment.

**Comment by emdupre** [comment]
There are a few minor corrections in references that would significantly improve the formatting of the paper.

- **Updating inclusion of multiple references:** For example,  this reference list on L13:
  _"...microsecond temporal resolution over arbitrarily long measurement times [@\lindfors2004PRL], [@\taylor2019interferometric], [@\RichardBook]. "_
  should be updated as:
  _"...microsecond temporal resolution over arbitrarily long measurement times [@\lindfors2004PRL; @\taylor2019interferometric; @\RichardBook]."_
- **Conflicting bracket-parentheses placement on L45:** Here,
  _"...such as lipid vesicles across the surface of giant unilamellar vesicles (GUVs) [@\spindler2018high](see \autoref{fig:piscat}a-(IV))"_
  the missing space between the Spindler and figure reference is yielding inappropriate formatting. I would recommend updating as follows:
  _"...such as lipid vesicles across the surface of giant unilamellar vesicles (GUVs; @\spindler2018high; see also \autoref{fig:piscat}a-(IV))"_

**Comment by po60nani** [comment]
@emdupre, Thank you for the comprehensive explanation. We changed the manuscript based on your explanation and comments regarding reference corrections. 

- We added the following sentence to clarify the PiSCAT dependencies:

_`PiSCAT uses a set of dependencies (e.g. PyQtGraph, scikit-learn/image [@scikit-learn], Trackpy [@Trackpy], and Astropy [@astropy:2018]) that assist us in achieving various goals. The PiSCAT repository contains this list.`_

- There are a few minor corrections in references that would significantly improve the formatting of the paper.  
 _`Done`_

**Comment by po60nani** [comment]
@editorialbot generate pdf

**Comment by emdupre** [comment]
Thank you, @po60nani. 

Please let us know, @aquilesC, if these edits are in line with what you were expecting. Please let me know, too, if my explanation helped to clarify your questions !

**Comment by aquilesC** [comment]
Thanks a lot @emdupre for your explanation, and @po60nani for the updated manuscript. 

I have checked the text and marked all the ticks for the review process as completed. I think the work is ready to be published.

**Comment by emdupre** [comment]
Thank you, @aquilesC ! Confirming that I've spoken to @ziatdinovmax via email, who has also confirmed that he's happy to recommend PiSCAT for publication in JOSS. Thank you both for your thoughtful reviews !

I'll perform a few last checks now, but at this point, @po60nani, could you please:

- [ ] Make a tagged release of your software, and list the version tag of the archived version here. 
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata (e.g., the JOSS paper and software archive have the same title and author listing)
- [ ] List the DOI of the archived version here.

I can then move forward with accepting the submission :rocket:

**Comment by emdupre** [comment]
@editorialbot check references

**Comment by po60nani** [comment]
Thank you @emdupre.  We updated the repository:

-  **Make a tagged release of your software, and list the version tag of the archived version here.**
  _PiSCAT version 0.1.23 ([link](https://github.com/SandoghdarLab/PiSCAT/releases/tag/0.1.23))_

-  **Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)**
 _Done!_

-  **Check the archival deposit (e.g., in Zenodo) has the correct metadata (e.g., the JOSS paper and software archive have the same title and author listing)**
 _Done! Here you can find the [Zendo](https://zenodo.org/record/6389882#.YkGRRtIzZaQ)_

-  **List the DOI of the archived version here.**

1. [DOI: 10.5281/zenodo.6389603](https://doi.org/10.5281/zenodo.6389603)
2. [DOI: 10.5281/zenodo.6389882](https://doi.org/10.5281/zenodo.6389882)

**Comment by emdupre** [comment]
Thank you, @po60nani ! So, to confirm, [10.5281/zenodo.6389882](https://doi.org/10.5281/zenodo.6389882) is the archive for the latest release. I will only associate one DOI with the JOSS publication, so please let me know if the other, earlier archive has non-overlapping information.

On the archive itself, a few formatting requests:
- [ ] Could you please update the title of the archive to match the software paper ? Currently, the archive title appears cut-off (i.e., is missing the term "microscopy" compared to the software paper). 
- [ ] Similarly, could you confirm the author listing in the software archive ? The first two authors are differently listed between the two objects (e.g., the first author is `Houman` in the software archive and `Houman Mirzaalian Dastjerdi` in the software paper). 

These edits can be made directly on the archived version, without minting a new DOI ! A guide is available [here](https://help.zenodo.org/#technical).

**Comment by emdupre** [comment]
@editorialbot set 0.1.23 as version

**Comment by emdupre** [comment]
@editorialbot set 10.5281/zenodo.6389882 as archive

**Comment by po60nani** [comment]
@emdupre, I apologize for the mistakes; they have been corrected.

**Comment by emdupre** [comment]
@editorialbot recommend-accept

**Comment by kyleniemeyer** [comment]
Hi @po60nani, I'm doing some final checks before publishing. Just a few things:
- in the statement of need section, could you change "e.g" to "e.g.," ?
- The journal name in the Dastjerdi et al. reference looks off, could you correct that?

**Comment by po60nani** [comment]
Hi, @kyleniemeyer. I fixed the problems.

**Comment by kyleniemeyer** [comment]
@editorialbot generate pdf

**Comment by kyleniemeyer** [comment]
@po60nani thanks—but unfortunately there is another issue that I missed, the Astropy paper is missing the journal name (we see this a lot because .bib files often have journal name macros that our system doesn't interpret). Could you fix that too?

**Comment by po60nani** [comment]
@kyleniemeyer, Thank you, I fixed it.

**Comment by kyleniemeyer** [comment]
@editorialbot accept
